### PR TITLE
[C++] Support Pointer to Integer casts

### PR DESCRIPTION
**Pull request by CyberChrisDE**
For clang, uintptr_t is just a typedef that may be defined by the programmer itself. To support calculations with pointer values anyway I added support for casts from pointers to integer types. The Verifast core supported this for integers with PtrRank only so far. So I had to add support for those casts more generally.

Please have a closer look at my added line in verifast1.ml to be sure that my changes are sound.
The removed line in verifast1.ml had an exact duplicate some lines below.

### CODE REVIEW

File: src/cxx_frontend/stubs/stubs_ast.capnp [hunk 1]
```diff
@@ -446,6 +446,7 @@ struct Expr {
     operatorCall @19 :Call;
     cleanups @20 :ExprNode;
     bindTemporary @21 :ExprNode;
+    integralCast @22 :StructToStruct;
```
Comment (btj) [inline src/cxx_frontend/stubs/stubs_ast.capnp:449]: Rename `StructToStruct`? Perhaps `Cast`?
```suggestion
    integralCast @22 :Cast;
```

**Review by btj — CHANGES_REQUESTED** [review]
Looks great! Just one suggestion.

**Comment by btj** [comment]
Interestingly, the Windows build fails with the following error message:
```
> D:/a/verifast/verifast/tests/cxx/casts.cpp(5,26): cast from pointer to smaller type 'unsigned long' loses information
```
Apparently, clang uses the host architecture as the target architecture, rather than whatever is specified on the VeriFast command line. For purposes of this PR, perhaps work around it by using `unsigned long long` instead of `unsigned long` in the test case. We might want to investigate this host dependency in a separate issue/PR (if someone cares enough).

**Comment by CyberChrisDE** [comment]
Type 'StuctToStruct' is used at some points already. As it should be the same for all casts, I changed its name to 'Cast'.
As you correctly stated, the different size of 'unsigned long' on Windows and Linux was the problem. I simply forgot to consider compilation/verification on Windows. It's fixed now.
For me there is no need for opening an issue. For now I don't really have a need for Windows support at all. Maybe at some point in the future. If someone wishes to have cross platform support, I'd have a look if I'm able to do the necessary changes.

**Review by NielsMommen — APPROVED** [review]
Looks good to me!

**Review by NielsMommen — APPROVED** [review]

**Review by btj — APPROVED** [review]

**Pull request merged**